linked_list/linkedList.c.c

```diff
@@ -0,0 +1,255 @@
+		while (temp->next != NULL) {	// Loop until we get last node
+			temp = temp->next;
+		}
+		temp->next = nw_node;		// Assiging address of nw_node to next variable of last element of list
+	}
+	return 1;
+}
+
+//  Adding at given position
+int addAtPos(struct Node **start, struct Node* nw_node, int pos) {
+	struct Node* temp;		// 'temp' temporary node to traverse to last.
+	if (*start == NULL) {
+		*start = nw_node;
+		return 1;
+	} else {
+		int i = 1;
+		temp = *start;
+		while (temp->next != NULL && i < pos-1) {	// Loop we reach to the position or last element of list is encountered
+			temp = temp->next;
+			i++;
+		}
+		if (i != pos-1) {
+			return 0;
+		} else {
+			nw_node->next = temp->next;		// Placing address of next node to nw_node->next
+			temp->next = nw_node;		// Assiging address of nw_node in list after temp node.
+			return 1;
+		}
+	}
+}
+
+//  Deleting first element
+struct Node* deleteFirst(struct Node **start) {
+	if (*start == NULL) {
+		return *start;
+	} else {
+		struct Node* temp = *start;
+		*start = (*start)->next;		// Shifting 'start' to second node
+		return temp;
+	}
+}
+
+//  Deleting last element
+struct Node* deleteLast(struct Node **start) {
+	if (*start == NULL) {
+		return NULL;
+	} else {
+		struct Node* temp = *start;		// temp node to go through each node
+		if((*start)->next == NULL) {
+			return 	*start;
+		}
+		while (temp->next->next != NULL)	// Loop until second last element is encountered
+			temp = temp->next;
+		struct Node* tmp = temp->next;		// storing last element
+		temp->next = NULL;			// Removing last element
+		return tmp;
+	}
+}
+
+//  Delete element at given position
+struct Node* deleteAtPos(struct Node **start, int pos) {
+	if (*start == NULL) {
+		return NULL;
+	} else {
+		struct Node* temp = *start;		// temp node to go through each node
+		int i = 1;
+		while (temp->next->next != NULL && i < pos-1) {	// until we reach to the position or last element of list is encountered
+			temp = temp->next;
+			i++;
+		}
+		if (i != pos-1) {
+			printf("Invalid Index\n\n");
+			return NULL;
+			} else {
+			struct Node* tmp = temp->next;
+				temp->next = temp->next->next;	// Replacing address of position 'pos' with  node next to 'pos'
+			return tmp;
+		}
+	}
+}
+
+//  To traverse whole list
+void traverse(struct Node *start) {
+	printf("Linked List\n");
+	if(start == NULL) {
+		printf("Linked List is Empty");
+	} else {
+		struct Node* temp = start;		// temp node to go through each node
+		while (temp->next != NULL) {
+			printf(" %d ->", temp->d);
+			temp = temp->next;
+		}
+		printf("%d", temp->d);
+	}
+	printf("\n\n");
+}
+
+//  To search an element
+int search(struct Node *start, int data) {
+	struct Node* temp = start;
+	int found = 0, i = 1;
+	while (temp->next != NULL) {
+		if(temp->d == data) {
+			found = 1;
+			break;
+		}
+		i++;
+		temp = temp->next;
+	}
+	if (found != 0) {
+		return i;
+	} else {
+		return -1;
+	}
+}
+
+// to print search result
+void searchResult(int d, int pos) {
```

Comment:
These are not LinkedList methods but just convenience methods for your example. In that case, please proceed their names by underscore (_) so that it looks distinct.